artifact/lexer.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#ifndef MENDER_ARTIFACT_LEXER_HPP
+#define MENDER_ARTIFACT_LEXER_HPP
+
+#include <cstdint>
+#include <string>
+#include <vector>
+#include <cstdlib>
+#include <cstdio>
+#include <unordered_map>
+
+#include <system_error>
+
+#include <common/io.hpp>
+#include <common/log.hpp>
+#include <common/expected.hpp>
+#include <artifact/tar/tar.hpp>
+
+#include <artifact/sha/sha.hpp>
+
+#include <artifact/token.hpp>
+
+
+namespace mender {
+namespace artifact {
+namespace lexer {
+
+using namespace std;
+
+namespace log = mender::common::log;
+
+class Lexer {
+private:
+	std::shared_ptr<mender::tar::Reader> &tar_reader;
```

Comment:
Can't this be a simple `mender::tar::Reader &tar_reader` reference?

Reply:
I removed the reference to the tar_reader, and made it `_` suffixed however